uaclient/entitlements/tests/test_cis.py

```diff
@@ -0,0 +1,87 @@
+"""Tests related to uaclient.entitlement.base module."""
+
+import mock
+from io import StringIO
+
+from uaclient import config
+from uaclient.entitlements.cis import CISEntitlement
+from uaclient.testing.helpers import TestCase
+
+
+CIS_MACHINE_TOKEN = {
+    'machineSecret': 'blah',
+    'machineTokenInfo': {
+        'contractInfo': {
+            'resourceEntitlements': [
+                {'type': 'cis-audit'}]}}}
+
+
+CIS_RESOURCE_ENTITLED = {
+    'resourceToken': 'TOKEN',
+    'entitlement': {
+        'obligations': {
+            'enableByDefault': True
+        },
+        'type': 'cis-audit',
+        'entitled': True,
+        'directives': {
+            'aptURL': 'http://CIS',
+            'aptKey': 'APTKEY'
+        },
+        'affordances': {
+            'series': []   # Will match all series
+        }
+    }
+}
+
+
+class TestCISEntitlementCanEnable(TestCase):
+
+    @mock.patch('os.getuid', return_value=0)
+    def test_can_enable_true_on_entitlement_inactive(self, m_getuid):
+        """When operational status is INACTIVE, can_enable returns True."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', CIS_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-cis-audit', CIS_RESOURCE_ENTITLED)
+        entitlement = CISEntitlement(cfg)
+        # Unset static affordance container check
+        entitlement.static_affordances = ()
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            self.assertTrue(entitlement.can_enable())
+        self.assertEqual('', m_stdout.getvalue())
+
+
+class TestCISEntitlementEnable(TestCase):
+
+    @mock.patch('uaclient.util.subp')
+    @mock.patch('uaclient.util.get_platform_info')
+    @mock.patch('os.getuid', return_value=0)
+    def test_enable_configures_apt_sources_and_auth_files(
+            self, m_getuid, m_platform_info, m_subp):
+        """When entitled, configure apt repo auth token, pinning and url."""
+        m_platform_info.return_value = 'xenial'
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', CIS_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-cis-audit', CIS_RESOURCE_ENTITLED)
+        entitlement = CISEntitlement(cfg)
+        # Unset static affordance container check
+        entitlement.static_affordances = ()
+
+        with mock.patch('uaclient.apt.add_auth_apt_repo') as m_add_apt:
+            with mock.patch('uaclient.apt.add_ppa_pinning') as m_add_pinning:
+                self.assertTrue(entitlement.enable())
+
+        add_apt_calls = [
+            mock.call('/etc/apt/sources.list.d/ubuntu-cis-audit-xenial.list',
+                      'http://CIS', 'TOKEN', None, 'APTKEY')]
+
+        subp_apt_cmds = [
+            mock.call(['apt-get', 'update'], capture=True),
+            mock.call(['apt-get', 'install', 'ubuntu-cisbenchmark-16.04'])]
```

Comment:
If the package name changes, and I hope it does, will this start failing?

Reply:
If the package name changes, we need to release a new UA-client, currently packages being installed are a static hardcoded list of packages for fips/fips-updates  and cis, We could ask the ua-contract backend to surface a new 'packages' directive which would allow us to react to a dynamic package dependency change in the future. There is nothing that the PPA itself can tell us about what needs to be installed (or run post-ppa enablement)

Reply:
We can also bake in the option to look for a 'packages' directive in ua client now and use that instead of the static list if ua-contracts surfaces it in the future.